web_services/foxx/brightid/operations.js

```diff
@@ -14,15 +14,24 @@ const {
 
 const TIME_FUDGE = 60 * 60 * 1000; // timestamp can be this far in the future (milliseconds) to accommodate client/server clock differences
 
-const verifyUserSig = function(message, id, sig) {
+const verifyUserSig = function(message, id, sig, opName) {
   const user = db.loadUser(id);
-  // this will happen for "Add Connection" when one party is not created
-  // this also enable this version of code to be used by the old users collection
-  // for users that don't have signingKey
-  const signingKey = (user && user.signingKey) ? user.signingKey : urlSafeB64ToB64(id);
-  if (!nacl.sign.detached.verify(strToUint8Array(message), b64ToUint8Array(sig), b64ToUint8Array(signingKey))) {
-    throw 'invalid signature';
+  // When "Add Connection" is called on a user that is not created yet
+  // signingKey can be calculated from user's brightid
+  let signingKey = user ? user.signingKey : urlSafeB64ToB64(id)
+  const signingKeys = [signingKey];
+  // 'Add Subkey' and 'Remove Subkey' can only be signed using master key
+  if (! ['Add Subkey', 'Remove Subkey'].includes(opName)) {
+    const subkeys = (user && user.subkeys) || [];
+    signingKeys.push(...subkeys)
   }
```

Comment:
I would prefer that any subkey can remove any or all other subkeys or add another subkey
```suggestion
  const subkeys = (user && user.subkeys) || [];
  signingKeys.push(...subkeys);
```

Reply:
In fact it would be nice if all signing keys were on the same level. There wouldn't be a notion of signingKey and subkeys, just signing keys.

Social recovery would still reset the signing keys back to just one.

I realize this is different than what I wrote in this issue.  https://github.com/BrightID/BrightID-Node/issues/155 I apologize.

I think this will make it easier to use a second device as a recovery device if you lose the first one--or just to transfer to another device without using social recovery.